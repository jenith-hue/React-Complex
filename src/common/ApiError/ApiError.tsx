import {
  makeStyles,
  RACButton,
  RACModal,
  Typography,
} from '@rentacenter/racstrap';
import React from 'react';
import { ReactComponent as ApiErrorIcon } from '../../assets/img/apiErrorIcon.svg';
import { FULL_API_ERROR_MESSAGE } from '../../constants/constants';

export interface ModalProps {
  open: boolean;
  message?: string;
  onClose: () => void;
}
export const useStyles = makeStyles((theme: any) => ({
  dialogContent: {
    textAlign: 'left',
    height: theme.typography.pxToRem(400),
  },
  dialogRoot: {
    '& .MuiDialogContent-root': {
      padding: '1rem',
      flexDirection: 'column',
      display: 'flex',
      justifyContent: 'space-between',
      alignItems: 'center',
      height: theme.typography.pxToRem(100),
    },
    '& .MuiDialog-paperWidthSm': {
      maxWidth: theme.typography.pxToRem(500),
      maxHeight: theme.typography.pxToRem(363),
    },
    '& .MuiTypography-h4': {
      fontSize: '20px !important',
    },
    '& .MuiOutlinedInput-multiline': {
      padding: theme.typography.pxToRem(10),
    },
    '& .MuiInputBase-multiline': {
      margin: '8px 0px 0px 0px !important',
    },
    '& .MuiInputBase-input': {
      fontFamily: 'OpenSans-regular',
      fontSize: theme.typography.pxToRem(14),
    },
  },
  dialogActions: {
    paddingRight: theme.typography.pxToRem(15),
    paddingBottom: theme.typography.pxToRem(15),
    justifyContent: 'center',
  },
  apiErrorMessage: {
    fontSize: '1rem !important',
  },
}));

export interface ApiErrorModalContentProps {
  message?: string;
}
const ApiErrorModalContent = ({ message }: ApiErrorModalContentProps) => {
  const classes = useStyles();
  return (
    <>
      <ApiErrorIcon />
      <Typography variant="body2" className={classes.apiErrorMessage}>
        {message || FULL_API_ERROR_MESSAGE}
      </Typography>
    </>
  );
};

export const ApiErrorModal = ({ message, open, onClose }: ModalProps) => {
  const classes = useStyles();

  return (
    <RACModal
      isOpen={open}
      titleVariant="h4"
      classes={{
        dialogContent: classes.dialogContent,
        dialog: classes.dialogRoot,
        dialogActions: classes.dialogActions,
      }}
      maxWidth="sm"
      title=""
      content={<ApiErrorModalContent message={message} />}
      onClose={onClose}
      buttons={
        <>
          <RACButton variant="contained" color="primary" onClick={onClose}>
            OK
          </RACButton>
        </>
      }
    />
  );
};
