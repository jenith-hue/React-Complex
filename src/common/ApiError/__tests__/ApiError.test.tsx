/* eslint-disable react/display-name */
/* eslint-disable sonarjs/no-identical-functions */
import React from 'react';
import renderer from 'react-test-renderer';

import { ApiErrorModal } from '../ApiError';

interface PropWithChildren {
  children: JSX.Element;
}
jest.mock('@rentacenter/racstrap', () => ({
  ...jest.requireActual('@rentacenter/racstrap'),
  Typography: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="Typography" {...rest}>
      {children}
    </div>
  ),
  RACButton: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="RACButton" {...rest}>
      {children}
    </div>
  ),
  RACModal: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="RACModal" {...rest}>
      {children}
    </div>
  ),
  makeStyles: () => () => ({
    dialogContent: 'dialogContent',
    dialogRoot: 'dialogRoot',
    dialogActions: 'dialogActions',
    apiErrorMessage: 'apiErrorMessage',
  }),
}));
describe('ApiErrorModal', () => {
  it('should render correctly', async () => {
    const tree = renderer.create(<ApiErrorModal open onClose={jest.fn()} />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
