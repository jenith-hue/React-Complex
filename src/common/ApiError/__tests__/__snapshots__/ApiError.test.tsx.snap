// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiErrorModal should render correctly 1`] = `
<div
  buttons={
    <React.Fragment>
      <RACButton
        color="primary"
        onClick={[MockFunction]}
        variant="contained"
      >
        OK
      </RACButton>
    </React.Fragment>
  }
  classes={
    Object {
      "dialog": "dialogRoot",
      "dialogActions": "dialogActions",
      "dialogContent": "dialogContent",
    }
  }
  content={<ApiErrorModalContent />}
  data-testid="RACModal"
  isOpen={true}
  maxWidth="sm"
  onClose={[MockFunction]}
  title=""
  titleVariant="h4"
/>
`;
